### PR TITLE
[WIP] globbing

### PR DESCRIPTION
**Pull request by anouvel**
TODO:

- [x] support for path (`ls ../../*`)
- [x] refactoring
- [ ] check the status code of the context loading
- [ ] support of '\\*'
- [ ] change the moment of the globbing (ls "*" should display cannot acces *)

### CODE REVIEW

**Comment by anouvel** [comment]
It works but this is a poc (leaks etc).

**Comment by anouvel** [comment]
Still leak

File: srcs/parser/globbing/globbing_run.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	if (*pattern == '?')
+	{
+		return (*input && check_globbing(pattern + 1, input + 1));
+	}
+	else if (*pattern == '*')
+	{
+		return (check_globbing(pattern + 1, input) ||
+				(*input && check_globbing(pattern, input + 1)));
+	}
+	else
```
Comment (jgigault) [inline srcs/parser/globbing/globbing_run.c:39]: May be we will have soin a additional condition for token `[` :)

**Review by gabkk — COMMENTED** [review]
check returns of all your memory allocation functions before doing the norme, ft_strdup ft_strjoin ft_strplit

**Comment by anouvel** [comment]
Some snippets @gabkk !?

All the process is "called before" by the context loading and "called back" by add_node_to_list, every null pointers are handled by these functions, maybe I forgot one, but I don't think so...
BTW, I don't check the status of the loading context, my bad.

File: srcs/parser/globbing/globbing_run.c [hunk 2]
```diff
@@ -56,6 +56,23 @@ static char	*s_join_free(char *s1, char *s2)
 	return (ret);
 }
 
+typedef struct s_mylist
+{
+	struct s_mylist		*next;
+	char				*content;
+}					t_mylist;
+
+void globbing_bracket(t_mylist **list, char *input)
```
Comment (anouvel) [inline srcs/parser/globbing/globbing_run.c:65]: This is the function to fill.

Basic example to test:

``` bash
mkdir -p tmp/a tmp/b
touch tmp/a/aa tmp/a/aaa
touch tmp/b/bb tmp/b/bbb
touch tmp/Z
```

with the command

``` bash
bash-3.2$ ls tmp/*/*[a..b]
tmp/a/aa	tmp/a/aaa	tmp/b/bb	tmp/b/bbb
```

@gabkk

**Comment by jgigault** [comment]
OBSOLETE PR

**Pull request closed without merging**